packages/user/src/UserService.ts

```diff
@@ -29,32 +33,77 @@ export class UserService {
 	public async getAssignedWishCards(user: IUser): Promise<IWishCard[]> {
 		const agency = await this.agencyRepository.getAgencyByUserId(user._id);
 
-		// If user hadn't filled out agency info, redirect them to form
 		if (!agency) {
-			throw new Error(`Failed to get Agency for user ${user.email}`);
+			throw new UserError(`Failed to get Agency for user ${user.email}`);
 		}
 		return this.wishCardRepository.getWishCardsByAgencyId(agency._id);
 	}
 
-	public async updateUserData(
-		id: TypeObjectId<IUser>,
-		params: Partial<Omit<IUser, '_id'>>,
-	): Promise<IUser> {
+	public async updateUserData(id: TypeObjectId<IUser>, params: Partial<IUser>): Promise<IUser> {
 		return this.userRepository.updateUserById(id, params);
 	}
 
-	public async verifyUserEmail(hash: string): Promise<boolean> {
+	public async verifyUserEmail(hash: string): Promise<IUser> {
 		const user = await this.userRepository.getUserByVerificationHash(hash);
 
-		if (user) {
-			if (user.emailVerified) {
-				return false;
-			}
+		if (!user) {
+			throw new UserError('No user found for verification!');
+		}
 
+		if (!user.emailVerified) {
 			await this.userRepository.setUserEmailVerification(user._id, true);
```

Comment:
Since verifyUserEmail returns a user, shouldn't we update it on setUserEmailVerification call. Otherwise the user returned here would still have emailVerified to be  false, no? :thinking:

Reply:
yeah that's true, that should be updated